libs/dal/src/repositories/preferences/preferences.entity.ts

```diff
@@ -0,0 +1,48 @@
+import { ChannelTypeEnum } from '@novu/shared';
+import type { OrganizationId } from '../organization';
+import type { EnvironmentId } from '../environment';
+import type { SubscriberId } from '../subscriber';
+import { UserId } from '../user';
+import { ChangePropsValueType } from '../../types';
+
+type WorkflowOptionsPreferences = {
+  workflow: {
+    defaultValue: boolean;
+    readOnly: boolean;
+  };
+  channels: {
+    [key in (typeof ChannelTypeEnum)[keyof typeof ChannelTypeEnum]]?: {
+      defaultValue: boolean;
+      readOnly: boolean;
+    };
+  };
+};
```

Comment:
```suggestion
type PreferenceOptions = {
    defaultValue: boolean;
    readOnly: boolean;
};

type WorkflowOptionsPreferences = {
  workflow: PreferenceOptions;
  channels: Record<ChannelTypeEnum, PreferenceOptions>;
};
```

or potentially it would need to be `Partial<Record<ChannelTypeEnum, PreferenceOptions>>;` if it doesn't have to be fully populated each time